### PR TITLE
RDCC-1583: Adding component name to all application logs

### PR DESCRIPTION
**Pull request by AdilOoze**
**JIRA link (if applicable)**

https://tools.hmcts.net/jira/browse/RDCC-1583

**Change description**

Adding component name to all application logs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -99,4 +99,6 @@ feign:
 resendInterval: 60
 
 # sync time interval in minutes
-syncInterval: 60
\ No newline at end of file
+syncInterval: 60
+
+logging-component-name: "RD_User_Profile_API"
```
Comment (KOTLAPRA23) [inline src/main/resources/application.yaml:104]: remove ""

Comment (AdilOoze) [reply src/main/resources/application.yaml]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/037909c45ca61e983546303bf3e8729e773401bc

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java [hunk 2]
```diff
@@ -34,7 +38,7 @@ public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
                         }
                     }
                 } else {
-                    log.warn("FeignHeadConfiguration", "Failed to get request header!");
+                    log.warn(loggingComponentName, "FeignHeadConfiguration", "Failed to get request header!");
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java:41]: can you make this consistent by adding {}:: everywhere, as done for other places ? are you updating {}:: for log.info only ?

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/advice/UserProfileControllerAdvice.java [hunk 3]
```diff
@@ -40,6 +40,9 @@
     @Value("${resendInterval}")
     private String resendInterval;
 
+    @Value("${logging-component-name}")
+    protected static String loggingComponentName;
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/advice/UserProfileControllerAdvice.java:44]: make this variable to private everywhere if not used by child class

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java (hunk 2 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/776a4b17788086cb843cb412cd6519e17bc09983

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/advice/UserProfileControllerAdvice.java (hunk 3 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/advice/UserProfileControllerAdvice.java]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/776a4b17788086cb843cb412cd6519e17bc09983

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java [hunk 4]
```diff
@@ -105,7 +109,7 @@ public AttributeResponse updateUserDetails(UpdateUserDetails updateUserDetails,
             Response response = idamClient.updateUserDetails(updateUserDetails, userId);
             responseEntity = JsonFeignResponseHelper.toResponseEntity(response, getResponseMapperClass(response, null));
         } catch (FeignException ex) {
-            log.error("SIDAM call failed:", ex);
+            log.error("{}::, {}, {}", loggingComponentName, "SIDAM call failed:", ex);
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java:112]: comma not required

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java]: Addressed in this commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/308f72cc18bb9e89d2c086c252d57bf1a17ebd6a

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ExceptionServiceImpl.java [hunk 5]
```diff
@@ -17,12 +18,15 @@
...
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
     public void throwCustomRuntimeException(ExceptionType className, String msg) {
         throwCustomRuntimeException(className, msg, HttpStatus.OK);
     }
 
     public void throwCustomRuntimeException(ExceptionType className, String msg, HttpStatus httpStatus) {
-        log.error(msg);
+        log.error("{}::, {}", loggingComponentName, msg);
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/ExceptionServiceImpl.java]: Addressed in this commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/308f72cc18bb9e89d2c086c252d57bf1a17ebd6a

File: src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java [hunk 6]
```diff
@@ -34,7 +38,7 @@ public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
                         }
                     }
                 } else {
-                    log.warn("FeignHeadConfiguration", "Failed to get request header!");
+                    log.warn("{}::, {}, {}", loggingComponentName, "FeignHeadConfiguration", "Failed to get request header!");
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/userprofileapi/config/FeignInterceptorConfiguration.java]: Addressed in this commit: https://github.com/hmcts/rd-user-profile-api/pull/395/commits/308f72cc18bb9e89d2c086c252d57bf1a17ebd6a

**Pull request merged**